lib/Runtime/Library/DateImplementation.h

```diff
@@ -382,6 +382,39 @@ namespace Js {
         return GetDateDefaultString<StringBuilder>(&ymd, &tzd, DateTimeFlag::None, scriptContext, newStringBuilder);
     }
 
+    const auto ConvertUInt32ToString_ZeroPad_4 = [](const uint32 value, char16 *const buffer, const CharCount charCapacity)
+    {
+        Assert(charCapacity >= 4);
+        if (value < 10)
+        {
+            buffer[0] = _u('0');
+            buffer[1] = _u('0');
+            buffer[2] = _u('0');
+            buffer[3] = static_cast<uint16>(value) + _u('0');
+            buffer[4] = 0;
+        }
+        else if (value < 100)
+        {
+            buffer[0] = _u('0');
+            buffer[1] = _u('0');
+            buffer[2] = (static_cast<uint16>(value) / 10) + _u('0');
+            buffer[3] = (static_cast<uint16>(value) % 10) + _u('0');
+            buffer[4] = 0;
+        }
+        else if (value < 1000)
+        {
+            buffer[0] = _u('0');
+            buffer[1] = (static_cast<uint16>(value) / 100) + _u('0');
+            buffer[2] = ((static_cast<uint16>(value) / 10) % 10) + _u('0');
+            buffer[3] = (static_cast<uint16>(value) % 10) + _u('0');
+            buffer[4] = 0;
+        }
+        else
+        {
+            const errno_t err = _ultow_s(value, buffer, charCapacity, 10);
+            Assert(err == 0);
+        }
+    };
```

Comment:
nit: newline after this

Reply:
In general the above function is a bit ugly but I understand the need to implement ~ this way.

Reply:
Fixed.

Reply:
It wasn't super clear from the read that the reason for converting the input number to uint16 is so that it is compatible with the char16 _u('0'). Can you add a comment explaining? What will behavior look like for years > 4 digits and > MAX_UINT16 ?

Reply:
Agreed. I fixed it by moving the `static_cast` to where the int32 value is assigned to char16, for readability and avoided > MAX_UINT16 issue.